torch/_inductor/select_algorithm.py

```diff
@@ -1122,218 +1122,219 @@ def __call__(
-                return
-
-            if (
-                precompilation_timeout_seconds is None
-                or precompilation_timeout_seconds <= 0
-            ):
-                return no_op
-
-            env_workers = get_env_num_workers()
-            num_workers = env_workers if env_workers is not None else (len(choices))
-
-            if num_workers <= 0:
-                return no_op
-
-            # https://github.com/python/cpython/issues/106905
-            if (
-                sys.version_info.major == 3
-                and sys.version_info.minor == 11
-                and sys.version_info.micro <= 8
-            ):
-                return no_op
+        def autotune(wait_on_precompile):
+            start_time = time.perf_counter()
+            wait_on_precompile()
+            end_time = time.perf_counter()
+            wait_on_precompile_time = end_time - start_time
 
-            # check local and global cache before precompiling
-            timings = self.lookup(
+            start_time = time.perf_counter()
+            choice_to_timing = self.lookup(
                 choices,
                 name,
                 inputs_key,
-                benchmark=None,
+                lambda: make_benchmark_fn()(choices),
             )
+            end_time = time.perf_counter()
+            autotuning_time = end_time - start_time
 
-            if timings:
-                return no_op
+            if (
+                (choice_to_timing == {})
+                or all([not math.isfinite(timing) for timing in choice_to_timing.values()])
+            ):
+                raise NoValidChoicesError
 
-            if config.search_autotune_cache and not (
-                config.max_autotune or config.max_autotune_gemm
+            if (
+                (make_benchmark_fn.cache_info().currsize > 1)
+                or (log.getEffectiveLevel() == logging.DEBUG)
+                or (config.trace.log_autotuning_results)
             ):
-                return no_op
+                self.log_results(
+                    name, input_nodes, choice_to_timing, autotuning_time, wait_on_precompile_time
+                )
 
-            precompile_key = (
-                f"{name}: {inputs_key} : {torch.get_float32_matmul_precision()}"
-            )
-            if precompile_func := self.precompile_cache.get(precompile_key):
-                return precompile_func
+            return choice_to_timing
 
-            log.info(
-                "Multithreaded precompilation for %d choices using %d worker threads",
-                len(choices),
-                num_workers,
+        wait_on_precompile = precompile(choices)
+        
+        if return_multi_template:
+            return torch._inductor.ir.TensorBox.create(
+                torch._inductor.ir.MultiTemplateBuffer(
+                    layout,
+                    input_nodes,
+                    autotune(wait_on_precompile),
+                )
```

Comment:
don't "prune" the choices anymore (used to remove all but the best ExternKernelCaller), from looking at the scheduler bit that handles this I don't think it was needed?